decidim-elections/app/controllers/decidim/elections/elections_controller.rb

```diff
@@ -39,6 +39,13 @@ def single
         elections.first if single?
       end
 
+      # Public: Checks if the user has already casted at least one vote for the current election.
+      #
+      # Returns Boolean.
+      def already_voted?
+        @already_voted ||= Decidim::Elections::Votes::UserElectionVotes.new(current_user, election).query.any?
```

Comment:
Should we consider only accepted votes?

Reply:
This is a good question, actually...
As a user, once I voted I would expect the system to tell me that I have voted. If it doesn't, I would find it confusing and possibly think that the system does not work properly.

On the other hand, if the vote was rejected for whatever reason and the system tells me I have voted, the user might just miss the fact that they voted unless they actively verify the vote.